autoload/neomake/compat.vim

```diff
@@ -9,3 +9,56 @@ else
     return get(getbufvar(a:buf, ''), a:key, a:def)
   endfunction
 endif
+
+if exists('*json_decode')
+  let neomake#compat#json_true = v:true
+  let neomake#compat#json_false = v:false
+  let neomake#compat#json_null = v:null
+
+  function! neomake#compat#json_decode(json) abort
+    return json_decode(a:json)
+  endfunction
+else
+  let neomake#compat#json_true = 1
+  let neomake#compat#json_false = 0
+  let neomake#compat#json_null = {}
```

Comment:
Though `json_true` and `json_false` is good enough in terms of compatibility, I think `json_null` could lead to false positives.  By using `{}`, a distinction between an empty JSON object and `null` would be impossible.

It might be better to set its value to a script-local function can't be duplicated by accident:

```vim
function! s:json_null()
endfunction

let neomake#compat#json_null = [function('s:json_null')]
```

This is along the lines of using `null = object()` in Python.

With this, there'd only be one "true" `null` value for the compat `json_decode()`.  The same would need to be done for `json_true` and `json_false` if there's ever a need to create a compat `json_encode()`.